lib/engine.js

```diff
@@ -390,6 +390,25 @@ class GameEngine {
     return JSON.parse(state);
   }
 
+  setColors(colors) {
+    if (colors.length !== this.width * this.height) {
+      throw new Error("Dimension mismatch");
+    }
+    this.invalidateCache();
+    this.time = 0;
+    const state = JSON.parse(this.initialState);
```

Comment:
This should be at the first line (as all local variable declarations should be), so you can read width and height from it.